activerecord/test/cases/callbacks_test.rb

```diff
@@ -476,4 +476,27 @@ def test_inheritance_of_callbacks
     child.save
     assert child.after_save_called
   end
+
+  def test_on_isnt_allowed
+    exception = assert_raises ArgumentError do
+      Class.new(ActiveRecord::Base) do
+        before_save(on: :create) {}
+      end
+    end
+    assert_equal exception.message, "Unknown key: :on. Valid keys are: :if, :unless, :prepend"
+
+    exception = assert_raises ArgumentError do
+      Class.new(ActiveRecord::Base) do
+        around_save(on: :create) {}
+      end
+    end
+    assert_equal exception.message, "Unknown key: :on. Valid keys are: :if, :unless, :prepend"
+
+    exception = assert_raises ArgumentError do
+      Class.new(ActiveRecord::Base) do
+        after_save(on: :create) {}
+      end
+    end
+    assert_equal exception.message, "Unknown key: :on. Valid keys are: :if, :unless, :prepend"
```

Comment:
These assertions are backwards: the expected value should be passed first, and the actual value second.

Reply:
Oh, okay. I'm usually use Rspec, which AFAIK tends to follow the opposite pattern.

Is there a style guide documenting things like this?

Reply:
@seanlinsley It's an xUnit idiom. All assertion methods are of the form `assert(expected, actual)`. Rspec just does it the other way, for the semantics of their api, eg `expect(actual).to be(expected)`. FWIW, some JS test suites also use the order actual, expected. 😮